cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -321,28 +322,86 @@ impl CoreEnvironment<ReadOnly> {
+        let lockfile: Lockfile = self.lock(flox)?.into();
+        let packages_in_includes = if let Some(compose) = lockfile.compose {
+            Self::get_includes_for_packages(&packages, &compose.include)?
+        } else {
+            HashMap::new()
+        };
+
+        let install_ids = match Self::get_install_ids(&self.manifest()?, packages) {
+            Ok(ids) => ids,
+            Err(err) => {
+                if let CoreEnvironmentError::PackageNotFound(ref package) = err {
+                    if let Some(include) = packages_in_includes.get(package) {
+                        return Err(EnvironmentError::Core(
+                            CoreEnvironmentError::PackageOnlyIncluded(
+                                package.to_string(),
+                                include.name.clone(),
+                            ),
+                        ));
+                    }
+                };
+                return Err(EnvironmentError::Core(err));
+            },
+        };
 
         let toml = remove_packages(&current_manifest_contents, &install_ids)
             .map_err(CoreEnvironmentError::ModifyToml)?;
         let (store_path, _) = self.transact_with_manifest_contents(toml.to_string(), flox)?;
+
         Ok(UninstallationAttempt {
             new_manifest: Some(toml.to_string()),
+            still_included: packages_in_includes,
             built_environment_store_paths: Some(store_path),
         })
     }
 
-    fn get_install_ids_to_uninstall(
+    /// Get the highest priority included environment which provides each package.
+    /// Packages that are not provided by any included environments will be absent from the map.
+    fn get_includes_for_packages(
+        packages: &[String],
+        includes: &[LockedInclude],
+    ) -> Result<HashMap<String, LockedInclude>, EnvironmentError> {
+        let mut result = HashMap::new();
+        for package in packages {
+            if let Some(include) = Self::get_include_for_package(package, includes)? {
+                result.insert(package.clone(), include);
+            }
+        }
+
+        Ok(result)
+    }
+
+    /// Detect which included environment, if any, provides a given package.
+    fn get_include_for_package(
+        package: &String,
+        includes: &[LockedInclude],
+    ) -> Result<Option<LockedInclude>, EnvironmentError> {
+        // Reverse of merge order so that we return the highest priority match.
+        for include in includes.iter().rev() {
+            match Self::get_install_ids(&include.manifest, vec![package.to_string()]) {
+                Ok(_) => return Ok(Some(include.clone())),
+                Err(CoreEnvironmentError::PackageNotFound(_)) => continue,
+                Err(CoreEnvironmentError::MultiplePackagesMatch(_, _)) => continue,
+                Err(err) => return Err(EnvironmentError::Core(err)),
+            }
+        }
+
+        Ok(None)
+    }
+
+    fn get_install_ids(
```

Comment:
I dont want to duplicate the whole logic, but _some_ hint as to what this function is for would be nice

```suggestion
    /// Resolve "loose" package references (e.g. pkg-paths),
    /// to `install_ids` if unambiguous
    /// so that installation references remain valid for other package operations.
    fn get_install_ids(
```